specification/loadtestservice/LoadTestService/routes.tsp

```diff
@@ -0,0 +1,488 @@
+import "@azure-tools/typespec-azure-core";
+import "@typespec/rest";
+import "@typespec/http";
+import "@typespec/openapi";
+import "./models.tsp";
+import "@azure-tools/typespec-autorest";
+
+using Azure.Core;
+using Azure.Core.Traits;
+using TypeSpec.Rest;
+using TypeSpec.Http;
+using TypeSpec.Versioning;
+using Autorest;
+using OpenAPI;
+
+namespace Microsoft.LoadTestService;
+
+@tag("Test")
+interface LoadTestAdministration {
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations"
+  @summary("Create a new test or update an existing test by providing the test Id.")
+  @doc("Create a new test or update an existing test by providing the test Id.")
+  @route("/tests/{testId}")
+  @patch
+  createOrUpdateTest is Azure.Core.Foundations.Operation<
+    {
+      @header
+      @doc("Content type.")
+      contentType: "application/merge-patch+json";
+
+      @doc("""
+Unique Id for the load test, must contain only lower-case alphabetic,
+numeric, underscore or hyphen characters.
+""")
+      @path
+      @maxLength(50)
+      @minLength(2)
+      @pattern("^[a-z0-9_-]*$")
+      testId: string;
+
+      @doc("Load test model")
+      @body
+      body: Test;
+    },
+    ResourceCreatedOrOkResponse<Test>
+  >;
+
+  @summary("Delete a test by its test Id.")
+  @doc("Delete a test by its test Id.")
+  deleteTest is StandardResourceOperations.ResourceDelete<Test>;
+
+  @summary("Get load test details by test Id")
+  @doc("Get load test details by test Id")
+  getTest is StandardResourceOperations.ResourceRead<Test>;
+
+  @summary("""
+Get all load tests by the fully qualified resource Id e.g
+subscriptions/{subId}/resourceGroups/{rg}/providers/Microsoft.LoadTestService/loadtests/{resName}.
+""")
+  @doc("""
+Get all load tests by the fully qualified resource Id e.g
+subscriptions/{subId}/resourceGroups/{rg}/providers/Microsoft.LoadTestService/loadtests/{resName}.
+""")
+  listTests is StandardResourceOperations.ResourceList<
+    Test,
+    ListQueryParametersTrait<ListTestQueryParams>
+  >;
+
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations"
+  #suppress "@azure-tools/typespec-azure-core/byos" //It is limiting for users to have storage first before creating a load test also it will be breaking change for us.
+  @summary("""
+Upload input file for a given test Id. File size can't be more than 50 MB.
+Existing file with same name for the given test will be overwritten. File
+should be provided in the request body as application/octet-stream.
+""")
+  @doc("""
+Upload input file for a given test Id. File size can't be more than 50 MB.
+Existing file with same name for the given test will be overwritten. File
+should be provided in the request body as application/octet-stream.
+""")
+  @route("/tests/{testId}/files/{fileName}")
+  @put
+  uploadTestFile is Azure.Core.Foundations.Operation<
```

Comment:
Why can't this be `ResourceOperations.ResourceCreateOrReplace`?

Reply:
This is because creating a file requires content of file as body and response is test file information (json). So, since the input and output is customized in case of put, therefore using generic template for this one as well.

Reply:
Following it with this issue - https://github.com/Azure/typespec-azure/issues/3743